registry/storage/driver/testsuites/testsuites.go

```diff
@@ -1224,3 +1239,27 @@ func firstPart(filePath string) string {
 		filePath = dir
 	}
 }
+
+type errReader struct {
+	n   int64
+	msg string
+}
+
+func (er *errReader) Read(p []byte) (int, error) {
+	toread := int64(len(p))
```

Comment:
Ok, so the invariant is to return an error after n bytes that is not an `io.EOF`?